src/main/java/com/nvidia/spark/rapids/jni/ParquetFooter.java

```diff
@@ -202,55 +202,27 @@ public static ParquetFooter readAndFilter(HostMemoryBuffer buffer,
     int parentNumChildren = schema.children.length;
     ArrayList<String> names = new ArrayList<>();
     ArrayList<Integer> numChildren = new ArrayList<>();
-    depthFirstNames(schema, ignoreCase, names, numChildren);
+    ArrayList<Integer> tags = new ArrayList<>();
```

Comment:
This smells an awful lot like a struct. I assume we have a benefit to keeping this in discrete arrays, but wanted to make sure.

Reply:
Java does not have structs. They have objects. The only way to access members of these objects in C/CPP involves a lot of JNI code. Doing it efficiently is even more complicated where you have to get the method/field ID of what you want to access, and even then it is slow compared to doing it this way.  Oddly this is faster and more efficient then trying to reach back into the JVM to do the processing.